bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,234 @@
+            } catch (Exception e) {
+                updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+                logger.warn("Error during starting the WebSocket polling Thread {}", e.toString());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public void dispose() {
+        updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+        pollingJob.cancel(true);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (YIODOCKRECEIVERSWITCH.equals(channelUID.getIdWithoutGroup())) {
+            if (YIOREMOTEHANDLESTATUS_actualstatus.equals(YIOREMOTEHANDLESTATUS.AUTHENTICATED)) {
+                if (command instanceof RefreshType) {
+                    // TODO: handle data refresh
+
+                }
+
+                if (command.toString().equals("ON")) {
+                    logger.debug("YIODOCKRECEIVERSWITCH ON procedure: Switching IR Receiver on");
+                    YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.IRRECEIVERON, "");
+                } else if (command.toString().equals("OFF")) {
+                    logger.debug("YIODOCKRECEIVERSWITCH OFF procedure: Switching IR Receiver off");
+                    YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.IRRECEIVEROFF, "");
+                } else {
+                    logger.debug("YIODOCKRECEIVERSWITCH no procedure");
+                }
+
+            } else {
+                logger.warn("YIOremoteDockHandler not authenticated");
+            }
+
+        } else if (YIODOCKSENDIRCODE.equals(channelUID.getIdWithoutGroup())) {
+            if (YIOREMOTEHANDLESTATUS_actualstatus.equals(YIOREMOTEHANDLESTATUS.AUTHENTICATED)) {
+                if (command instanceof RefreshType) {
+                    // TODO: handle data refresh
+                    // logger.warn("YIOremoteDockHandler not authenticated");
+                }
+                logger.debug("YIODOCKSENDIRCODE procedure: {}", command.toString());
+                send_ircode = command.toString();
+                if (send_ircode.matches("[0-9][;]0[xX][0-9a-fA-F]+[;][0-9]+[;][0-9]")) {
+                    YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.IRSEND, send_ircode);
+                } else {
+                    logger.warn("Wrong IR code Format {}", send_ircode);
+                    send_ircode = "";
+                }
+            } else {
+                logger.warn("YIOremoteDockHandler not authenticated");
+            }
+        }
+    }
+
+    protected void updateChannelString(String group, String channelId, String value) {
+        ChannelUID id = new ChannelUID(getThing().getUID(), group, channelId);
+
+        if (isLinked(id)) {
```

Comment:
No harm in updating an unlinked Channel.

Reply:
I still see the `if`.

Reply:
Solved